### ISSUE TITLE
Make P2ArtifactRepositoryLayout actually lookup artifacts

### ISSUE THREAD

**Issue by laeubi** [eclipse/tycho#590]
currently the `P2ArtifactRepositoryLayout` only provides a disabled repository. We should implement it that way that it is possible to resolve `p2.` prefixed artifacts e.g in the form of:

```
<dependency>
  <groupId>p2.eclipse.plugin</groupId>
  <artifactId>org.eclipse.osgi</artifactId>
  <version>3.17.0.v20210823-1805</version>
  <type>eclipse-plugin</type>
</dependency>
```

if I add a repository of layout p2 it should be able to resolve this to the correct item.

**Comment by mickaelistria** [comment]
If we keep type `eclipse-plugin` or other tycho-specific types, we can just ignore the groupId and filter on the type.

**Comment by laeubi** [comment]
Yes that's the idea here. in the above example one would simply search for an IU with `id=org.eclipse.osgi` and `version= 3.17.0.v20210823-1805` of `namespace=osgi.bundle` (what we might can support as an alternative type name)

**Comment by laeubi** [comment]
Sadly maven does not support version-range resolving through custom layouts, I've opened https://issues.apache.org/jira/browse/MNG-7421

**Comment by laeubi** [comment]
We might be able to generate a https://maven.apache.org/ref/3.2.5/maven-repository-metadata/ for a P2 repository with tycho.
